cafebabel/static/js/edit-article.js

```diff
@@ -0,0 +1,147 @@
+class Tags {
+  constructor() {
+    const domContext = document.querySelector('.tags')
+    this.list = domContext.querySelector('.tags-list')
+    this.values = Array.from(this.list.querySelectorAll('input')).map(
+      input => input.value
+    )
+  }
+  addRemoveListener() {
+    const tagsButtonRemove = document.querySelectorAll('.tags .tags-list li a')
+    tagsButtonRemove.forEach(tagButtonRemove =>
+      Listener.clickRemove(tagButtonRemove)
+    )
+  }
+  checkSubmission(query) {
+    return query && !this._isTag(query)
+  }
+  add(query) {
+    this.values = this.values.concat(query)
+  }
+  remove(query) {
+    this.values = this.values.filter(value => value !== query)
+  }
+  display() {
+    const container = this.list.cloneNode(false)
+    this.list.replaceWith(this._createTagsList(container, this.values))
+    this.addRemoveListener()
+  }
+  _isTag(tag) {
+    return this.values.includes(tag)
+  }
+  _createTagsList(container, tagValues) {
+    tagValues.forEach((tagValue, index) =>
+      container.appendChild(this._createTag(tagValue, index))
+    )
+
+    return container
+  }
+  _createTag(tagValue, index) {
+    const li = document.createElement('li')
+    const input = document.createElement('input')
+    const a = document.createElement('a')
```

Comment:
I second @vinyll here, semantically it's closer to a `button` element. Thoughts?